projects/components/src/action-menu/action-search.provider.ts

```diff
@@ -0,0 +1,120 @@
+/*!
+ * Copyright 2020 VMware, Inc.
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+import { TranslationService } from '@vcd/i18n';
+import { ActionItem } from '../common/interfaces/index';
+import { SpotlightSearchProvider, SpotlightSearchResult, SpotlightSearchService } from '../spotlight-search/index';
+import { CommonUtil } from '../utils/index';
+
+const ACTION_PROVIDER_SECTION_TITLE_KEY = 'vcd.cc.action.provider.section.title';
+
+export class ActionSearchProvider<R, T> implements SpotlightSearchProvider {
+    private flatListOfAvailableActions: ActionItem<R, T>[] = [];
+
+    /**
+     * Initialized by the calling component and is used for searching of the search criteria entered in {@link SpotlightSearchComponent}
+     */
+    private _actions: ActionItem<R, T>[] = [];
+    set actions(actions: ActionItem<R, T>[]) {
+        this._actions = actions;
+        this.updateFlatListOfAvailableActions();
+    }
+    get actions(): ActionItem<R, T>[] {
+        return this._actions;
+    }
+
+    /**
+     * Used in {@link isActionDisabled} to calculate disabled state of actions
+     */
+    private _selectedEntities: R[] = [];
+    set selectedEntities(entities: R[]) {
+        this._selectedEntities = entities;
+        this.updateFlatListOfAvailableActions();
```

Comment:
Isn't it better to this lazily when doing the first search? Otherwise on any selection change, we will do extra computation even if not necessary